File: CPP/Case.cpp
// Case.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Case : public Komponen
{
private:
    string ukuran;  // ATX, Micro-ATX, Mini-ITX, dll
    string tipe;    // Tipe casing (Full Tower, Mid Tower, Mini Tower, dll)
public:
    Case() {}

    Case(string ukuran, string tipe, string merk, string nama) : Komponen(merk, nama)
    {
        this->ukuran = ukuran;
        this->tipe = tipe;
    }

    void setUkuran(string ukuran)
    {
        this->ukuran = ukuran;
    }

    void setTipe(string tipe)
    {
        this->tipe = tipe;
    }

    string getUkuran()
    {
        return this->ukuran;
    }

    string getTipe()
    {
        return this->tipe;
    }

    ~Case() {}
};

File: PYTHON/Case.py
class Case:
    def __init__(self, ukuran, tipe, merk, nama):
        self.ukuran = ukuran
        self.tipe = tipe
        self.merk = merk
        self.nama = nama

    def set_ukuran(self, ukuran):
        self.ukuran = ukuran

    def set_tipe(self, tipe):
        self.tipe = tipe

    def get_ukuran(self):
        return self.ukuran

    def get_tipe(self):
        return self.tipe

    def get_merk(self):
        return self.merk

    def get_nama(self):
        return self.nama
